raft/raft.go

```diff
@@ -575,7 +587,9 @@ func (r *raft) Step(m pb.Message) error {
 	case m.Term > r.Term:
 		lead := m.From
 		if m.Type == pb.MsgVote {
-			if r.checkQuorum && r.state != StateCandidate && r.electionElapsed < r.electionTimeout {
+			force := len(m.Entries) == 1 && bytes.Equal(m.Entries[0].Data, []byte(LeaderTransfer))
```

Comment:
I don't like using `Entries[0].Data` to store data in non-`MsgApp` messages. I'd rather introduce a top-level `Data` field in `raftpb.Message`.